src/test/java/com/polishuchenko/bookstore/service/CategoryServiceImplTest.java

```diff
@@ -0,0 +1,117 @@
+package com.polishuchenko.bookstore.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+import com.polishuchenko.bookstore.dto.category.CategoryDto;
+import com.polishuchenko.bookstore.dto.category.CategoryRequestDto;
+import com.polishuchenko.bookstore.exception.EntityNotFoundException;
+import com.polishuchenko.bookstore.mapper.CategoryMapper;
+import com.polishuchenko.bookstore.model.Category;
+import com.polishuchenko.bookstore.repository.category.CategoryRepository;
+import com.polishuchenko.bookstore.service.category.impl.CategoryServiceImpl;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryServiceImplTest {
+    private static final Long INVALID_ID = 99L;
+    private static final Long VALID_ID = 1L;
+    private static Category detective;
+    private static CategoryDto detectiveDto;
+    @Mock
+    private CategoryRepository categoryRepository;
+    @Mock
+    private CategoryMapper categoryMapper;
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+
+    @BeforeAll
+    public static void setUp() {
+        detective = new Category();
+        detective.setId(VALID_ID);
+        detective.setName("detective");
+
+        detectiveDto = new CategoryDto();
+        detectiveDto.setId(detective.getId());
+        detectiveDto.setName(detective.getName());
+    }
+
+    @Test
+    @DisplayName("Find all categories")
+    public void findAll_ReturnsListWithAllCategories() {
+        when(categoryRepository.findAll()).thenReturn(List.of(detective));
+        when(categoryMapper.toDto(ArgumentMatchers.any())).thenReturn(detectiveDto);
+
+        List<CategoryDto> actual = categoryService.findAll();
```

Comment:
If you implemented pagination to category service, provide pageble interface also

Reply:
I didn't implement pagination in category service